### ISSUE TITLE
pyright does not support else: with exceptions

### ISSUE THREAD

**Issue by lucidbee** [microsoft/pyright#132]
The function below returns a value on all paths but pyright does not understand that.

```
def foo(x, y) -> bool:
    try:
        z = x / y
    except Exception as e:
        return False
    else:
        return True
```

1:18 - error: Function with declared type of 'bool' must return value
1 error, 0 warnings

This function should not get this error.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version.

**Comment by erictraut** [comment]
This is now fixed in 1.0.30, which I just published.

I also added this to the unit tests to prevent future regressions.
